### DISCUSSION TITLE
Extra modulo character without endl using cout

### DISCUSSION THREAD

**Discussion by tridivb** [cpp-for-yourself/lectures-and-homeworks#32] [General]
I noticed a behaviour which struck me as weird while experimenting with cout. If the line is not ended explicitly at the last cout statement, it prints an extra modulo character by default. Why is that? Is there some form of overflow going on? This does not happen while using other print formats like printf and puts. Did I miss something about this in one of the earlier videos?

For reference, I was playing around with this code when I noticed it.

```
#include <iostream>
#include <vector>

using std::cin;
using std::cout;
using std::endl;
using std::vector;

[[nodiscard]] vector<int> CreateFibonacciSequence(const std::size_t length) {
  vector<int> sequence(length, 1);

  if (length < 3) { return sequence; }

  for (auto i = 2UL; i < length; i++) {
    sequence[i] = sequence[i - 1UL] + sequence[i - 2UL];
  }
  return sequence;
}

int main() {
  const vector<int> fib_seq = CreateFibonacciSequence(10UL);
  for (const auto& val : fib_seq) { cout << val << " "; }
  cout << endl;
  for (auto it = fib_seq.begin(); it != fib_seq.end(); it++) {
    cout << *it << " ";
  }
  cout << endl;
  for (int index = 0; index < fib_seq.size(); index++) {
    cout << fib_seq[index] << " ";
  }
  //   cout << endl;
  return 0;
}
```
![Screenshot from 2022-10-18 10-22-48](https://user-images.githubusercontent.com/28273134/196376985-1442fd36-594a-45b0-8958-87c6a29cd367.png)

**Comment by niosus** [comment]
This is indeed a bit weird, but here is what it is - it's just zsh telling you that there is no caret return symbol at the end of your program. See this answer for a bit more detail: https://unix.stackexchange.com/a/167600
